File: frontend/src/lib/api/schemas/pipelines.ts
import { z } from "zod";
import { JobStatus } from "@/types/enums";

const jobStatusSchema = z.enum([
  JobStatus.RUNNING,
  JobStatus.COMPLETED,
  JobStatus.FAILED,
  JobStatus.TERMINATED,
  JobStatus.TIMED_OUT,
  JobStatus.CANCELLED,
]);

export const jobReadSchema = z.object({
  id: z.string().uuid(),
  temporal_workflow_id: z.string(),
  temporal_run_id: z.string(),
  workflow_type: z.string(),
  project_id: z.string().uuid().nullable(),
  status: jobStatusSchema,
  started_at: z.string(),
  completed_at: z.string().nullable(),
  last_heartbeat_at: z.string().nullable(),
  error: z.string().nullable(),
});
export type JobRead = z.infer<typeof jobReadSchema>;

File: backend/src/ytforge/interfaces/api/schemas/pipelines.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from ytforge.domain.enums import JobStatus


class StartPipelineRequest(BaseModel):
    project_id: uuid.UUID
    topic: str


class StartPipelineResponse(BaseModel):
    workflow_id: str
    run_id: str


class JobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    temporal_workflow_id: str
    temporal_run_id: str
    workflow_type: str
    project_id: uuid.UUID | None
    status: JobStatus
    started_at: datetime
    completed_at: datetime | None
    last_heartbeat_at: datetime | None
    error: str | None
